packages/gestalt-design-tokens/build.js

```diff
@@ -393,12 +414,35 @@ StyleDictionary.registerFormat({
   },
 });
 
+StyleDictionary.registerTransform({
+  name: 'attribute/custom-cti',
+  type: 'attribute',
+  transformer(prop) {
+    const prefixes = ['base', 'sema', 'comp'];
+    const hasPrefix = prefixes.some((prefix) => prop.path[0] === prefix);
+
+    const attrNames = ['category', 'type', 'item', 'subitem', 'state'];
+    if (hasPrefix) {
+      attrNames.unshift('prefix');
+    }
+
+    const originalAttrs = prop.attributes || {};
+    const generatedAttrs = {};
+
+    for (let i = 0; i < prop.path.length && i < attrNames.length; i += 1) {
+      generatedAttrs[attrNames[i]] = prop.path[i];
```

Comment:
Honestly, I am not sure how this works. If `prop.path` and `attrNames` may have different `length`s, how do we know they both follow the same sequence? Because you're maching them with indices.

Reply:
@rlingineni also you said there were no major changes in the generated output code. Can you explaint how it works? 🤔